### DISCUSSION TITLE
URL space replace with - hyphen

### DISCUSSION THREAD

**Discussion by arirah** [vercel/vercel#8449] [Help]
How to URL space replace with - hyphen

example : `www.example.com/space in url` -> `www.example.com/space-in-url`

**Comment by amyegan** [comment]
Hi @arirah! The correct answer will depend on your project and which tools you are using. If you can share your repo or some more info about it, I would be happy to take a look.

**Reply by arirah** [reply]
> Hi @arirah! The correct answer will depend on your project and which tools you are using. If you can share your repo or some more info about it, I would be happy to take a look.

Hey @amyegan , thanks for your reply, I am trying to do this with nextjs in `next.config` file.

**Reply by amyegan** [reply]
I'm not aware of a `next.config.js` setting that would automatically replace spaces with dashes. Can you help me understand the scenario you are trying to solve? I'm linking some solutions to scenarios I could think of, but I know your specific case might be different.

- Visitor entered spaces instead of dashes in browser
  - Use [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) or [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) in `next.config.js` with regex to detect spaces and use [MIddleware](https://nextjs.org/docs/advanced-features/middleware) to re-route
  - Use [Dynamic Routes](https://nextjs.org/docs/routing/dynamic-routes) and [redirect](https://nextjs.org/docs/api-reference/data-fetching/get-static-props#redirect) when spaces are detected
- Trying to set a string as the path
  -  https://stackoverflow.com/a/70140809
  - https://stackoverflow.com/a/71620666

You might be able to find more solutions in [Next.js Discussions](https://github.com/vercel/next.js/discussions) or [Next.js Discord](http://nextjs.org/discord) as well.
